internal/httpproxy/store.go

```diff
@@ -0,0 +1,163 @@
+package httpproxy
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/pkg/errors"
+)
+
+// Mode definition
+type Mode string
+
+const (
+	// ModePassthrough mode
+	ModePassthrough Mode = "passthru"
+)
+
+// ErrorResponse definition
+type ErrorResponse struct {
+	Error string `json:"error"`
+}
+
+// Request definition
+type Request struct {
+	Method  string      `json:"method"`
+	Path    string      `json:"path"`
+	Headers http.Header `json:"header"`
+	Query   url.Values  `json:"query"`
+	Body    []byte      `json:"body"`
+}
+
+// WriteResponse definition
+type WriteResponse struct {
+	StatusCode int               `json:"statuscode"`
+	Headers    http.Header       `json:"header"`
+	Body       WriteResponseBody `json:"body"`
+}
+
+// WriteResponseBody definition
+type WriteResponseBody struct {
+	Offset int `json:"offset"`
+}
+
+// storeEntry definition
+type storeEntry struct {
+	Offset  int
+	Request Request
+}
+
+// StoreConfig definition
+type StoreConfig struct {
+	Mode Mode `json:"mode"`
+}
+
+// Store definition
+type Store struct {
+	Name string
+	Mode Mode
+	Data []storeEntry
+}
+
+// write stores incoming request data
+func (store *Store) write(w http.ResponseWriter, r *http.Request) {
+
+	var err error
+
+	reqData := Request{
+		Method:  r.Method,
+		Path:    r.URL.RequestURI(),
+		Headers: r.Header,
+		Query:   r.URL.Query(),
+	}
+
+	offset := len(store.Data)
+
+	resData := WriteResponse{
+		StatusCode: http.StatusOK,
+		Headers: http.Header{
+			"Content-Type": []string{"application/json"},
+		},
+		Body: WriteResponseBody{
+			Offset: offset,
+		},
+	}
```

Comment:
This makes no sense. It is unused. Below you only use `resData.Body`. Streamline this here! You can get rid of `WriteResponse`. You might need `WriteResponseBody`. But you need it only for response in line 101, so you can define the type in line.